addon/-private/utils/flatten.ts

```diff
@@ -24,7 +24,7 @@ export default function flatten<T>(src: NestedStructure<T>): Record<string, T> {
       const hash = flatten(value);
 
       for (const suffix in hash) {
-        result[`${key}.${suffix}`] = hash[suffix];
+        result[`${key}.${suffix}`] = hash[suffix]!;
```

Comment:
Here we could check that `hash[suffix]` is not `undefined`, but that would change which keys would be added to `result`.

We could update `result` to allow `undefined` values

```ts
const result = new EmptyObject() as Record<string, T | undefined>;
```

Line 30 has a similar issue

Reply:
we can check that `hash[suffix]` is not `undefined` as well